### PR TITLE
ARROW-13443 [C++]: Fix the incorrect mapping from flatbuf::MetadataVersion to arrow::ipc::MetadataVersion

### PR DESCRIPTION
**Pull request by liyafan82**
Please see https://issues.apache.org/jira/browse/ARROW-13443

### CODE REVIEW

File: cpp/src/arrow/ipc/metadata_internal.cc [hunk 1]
```diff
@@ -74,7 +74,7 @@ MetadataVersion GetMetadataVersion(flatbuf::MetadataVersion version) {
       return MetadataVersion::V2;
     case flatbuf::MetadataVersion::V3:
       // Arrow 0.3 to 0.7.1
-      return MetadataVersion::V4;
+      return MetadataVersion::V3;
```
Comment (kszucs) [inline cpp/src/arrow/ipc/metadata_internal.cc:77]: Nice catch! Do we have any relevant issue reports?

**Comment by kszucs** [comment]
@liyafan82 there are link errors in the windows builds.

**Review by pitrou — APPROVED** [review]
+1, thank you for noticing this @liyafan82

**Comment by liyafan82** [comment]
@kszucs @pitrou @nealrichardson Thanks a lot for your feedback and effort. 
I do not have an issue report, just happened to spot this.

**Pull request closed without merging**